fix: make TrustedDeviceTokenStorage conform to ResetInterface

dt-thomas-durand opened the pull request:
<!--
👍🎉 First off, thanks for taking the time to contribute! 🎉👍

Here are some tips for you:
 - Please follow the PR contribution guidelines: https://github.com/scheb/2fa/blob/7.x/CONTRIBUTING.md#creating-a-pull-request
 - Don't break backwards compatibility. If you have to, let's discuss! :)
 - Always add/update tests and ensure the build passes
-->

**Description**
<!--
Please provide a clear and concise description of the change.

For bug fixes:
 - What problem does the PR solve?
 - If this fixes an open issue, please link the bug ticket

For new features:
 - What's the motivation for this change? What's the problem you're trying to solve?
 - Why do you think it's a good idea to solve it this way?
-->
In certain scenarios, TrustedDeviceTokenStorage might not be re-instanciated between requests. That's the case when multiple requests are handled by a single worker, like when serving the application with RoadRunner or with FrankenPHP. In that case, the trusted_device cookie leak from a request to another, and break the feature entirely. By implementing ResetInterface, the storage is now properly reseted between requests

dt-thomas-durand committed:
fix: make TrustedDeviceTokenStorage conform to ResetInterface

In certain scenarios, TrustedDeviceTokenStorage might not be re-instanciated between requests.
That's the case when multiple requests are handled by a single worker, like when serving the application with RoadRunner or with FrankenPHP.
In that case, the trusted_device cookie leak from a request to another, and break the feature entirely.
By implementing ResetInterface, the storage is now properly reseted between requests

scheb commented:
Yes, we could add this. Two things that we need to change:

- Add a unit test to ensure the values are actually reset
- Please declare `symfony/contracts` as a dependency on the package

dt-thomas-durand commented:
> Yes, we could add this. Two things that we need to change:
> 
> 
> 
> - Add a unit test to ensure the values are actually reset
> 
> - Please declare `symfony/contracts` as a dependency on the package
> 
> 

Thanks for your feedback.
I'll add those asap!

dt-thomas-durand committed:
chore: added symfony/contract as an explicit dependency.

Added it to the bundle for maintainability reasons.
I'll be easier to upgrade Symfony if all Symfony dependencies lies in the bundle

dt-thomas-durand committed:
tests: added test coverage of TrustedDeviceTokenStorage->reset()

scheb commented:
Perfect, thank you very much! Released as v7.7.0.
